Gems/GradientSignal/Code/Tests/GradientSignalTestFixtures.cpp

```diff
@@ -115,6 +168,108 @@ namespace UnitTest
         }
     }
 
+    void GradientSignalBenchmarkFixture::RunSamplerGetValuesBenchmark(benchmark::State& state)
+    {
+        AZ_PROFILE_FUNCTION(Entity);
+
+        // All components are created, so activate the entity
+        ActivateEntity(m_testEntity.get());
+
+        // Create a gradient sampler and run through a series of points to see if they match expectations.
+        GradientSignal::GradientSampler gradientSampler;
+        gradientSampler.m_gradientId = m_testEntity->GetId();
+
+        // Get the height and width ranges for querying from our benchmark parameters
+        float height = aznumeric_cast<float>(state.range(0));
+        float width = aznumeric_cast<float>(state.range(1));
+        int64_t totalQueryPoints = state.range(0) * state.range(1);
+
+        // Call GetValues() for every height and width in our ranges.
+        for (auto _ : state)
+        {
+            // Set up our vector of query positions.
+            AZStd::vector<AZ::Vector3> positions(totalQueryPoints);
+            size_t index = 0;
+            for (float y = 0.0f; y < height; y += 1.0f)
+            {
+                for (float x = 0.0f; x < width; x += 1.0f)
+                {
+                    positions[index++] = AZ::Vector3(x, y, 0.0f);
```

Comment:
```
positions[index] = AZ::Vector3(x, y, 0.0f);
++index;
```
Is slightly faster since it doesn't need to increment then return the old value

Reply:
Both myself and https://godbolt.org/ respectfully disagree.  Both versions produce identical code on both Clang and MSVC.  It turns out compiler optimizers have improved over the past 30 years. ;)  I'll stick with the more compact version.  (I'm also honestly not worried about optimizing the test harness)